packages/rum-core/src/domain/foregroundContexts.spec.ts

```diff
@@ -0,0 +1,280 @@
+import { createNewEvent } from '../../../core/test/specHelper'
+import { InForegroundPeriod } from '../rawRumEvent.types'
+import { startForegroundContexts, ForegroundContexts } from './foregroundContexts'
+
+const FIVE_SECONDS_MS = 5_000 as Duration
+const TEN_SECONDS_MS = 10_000 as Duration
+
+describe('foreground', () => {
+  let foregroundContext: ForegroundContexts
+  let setupBuilder: TestSetupBuilder
+
+  beforeEach(() => {
+    setupBuilder = setup()
+      .withFakeClock()
+      .withConfiguration({ isEnabled: () => true })
+      .beforeBuild(({ configuration }) => {
+        foregroundContext = startForegroundContexts(configuration)
+        return foregroundContext
+      })
+  })
+
+  afterEach(() => {
+    setupBuilder.cleanup()
+  })
+
+  describe('when the page do not have the focus when starting', () => {
+    beforeEach(() => {
+      spyOn(Document.prototype, 'hasFocus').and.callFake(() => false)
+    })
+    describe('without any focus not blur event', () => {
+      describe('getInForeground', () => {
+        it('should false', () => {
+          const { clock } = setupBuilder.build()
+
+          clock.tick(1_000)
+
+          expect(foregroundContext.getInForeground(relativeNow())).toEqual({ view: { in_foreground: false } })
+        })
+      })
+
+      describe('getInForegroundPeriods', () => {
+        it('should an empty array', () => {
+          const { clock } = setupBuilder.build()
+
+          clock.tick(1_000)
+
+          expect(foregroundContext.getInForegroundPeriods(relativeNow(), 0 as ServerDuration)).toEqual([])
+        })
+      })
+    })
+
+    describe('with two closed focus period & one active every 5 seconds lasting 10 seconds', () => {
+      /* 
+      events         F      B   F       B   F
+      periods        <------>   <------->   <---- - - -
+      time       0   5  10  15  20  25  30  35  40  45
+      */
+      beforeEach(() => {
+        const { clock } = setupBuilder.build()
+        clock.tick(FIVE_SECONDS_MS)
+        window.dispatchEvent(createNewEvent('focus'))
+        clock.tick(TEN_SECONDS_MS)
+        window.dispatchEvent(createNewEvent('blur'))
+        clock.tick(FIVE_SECONDS_MS)
+        window.dispatchEvent(createNewEvent('focus'))
+        clock.tick(TEN_SECONDS_MS)
+        window.dispatchEvent(createNewEvent('blur'))
+        clock.tick(FIVE_SECONDS_MS)
+        window.dispatchEvent(createNewEvent('focus'))
+        clock.tick(TEN_SECONDS_MS)
+      })
+
+      describe('getInForeground', () => {
+        describe('after 2 second', () => {
+          it('should false', () => {
+            expect(foregroundContext.getInForeground(2_000 as RelativeTime)).toEqual({ view: { in_foreground: false } })
+          })
+        })
+
+        describe('after 10 second', () => {
+          it('should true', () => {
+            expect(foregroundContext.getInForeground(10_000 as RelativeTime)).toEqual({ view: { in_foreground: true } })
```

Comment:
it could be more helpful to explain what case you are testing with these values, rather than saying that you are testing these values.
what make 2 second and 10 second special?